HpToolsLauncher/Runners/FileSystemTestsRunner.cs

```diff
@@ -207,58 +210,51 @@ public class FileSystemTestsRunner : RunnerBase, IDisposable
                         }
                     }
                     //--handle mtb files (which contain links to tests)
-                    else
-                    //file might be LoadRunner scenario or
-                    //mtb file (which contain links to tests)
-                    //other files are dropped
+                    else //file might be LoadRunner scenario or mtb file (which contain links to tests) other files are dropped
                     {
-                        testGroup = new List<TestInfo>();
                         FileInfo fi = new FileInfo(source.Tests);
-                        if (fi.Extension == Helper.LoadRunnerFileExtention)
+                        if (fi.Extension == Helper.LoadRunnerFileExtension)
                             testGroup.Add(new TestInfo(source.Tests, source.Tests, source.Tests, source.Id)
                             {
                                 ReportPath = source.ReportPath
                             });
-                        else if (fi.Extension == ".mtb")
-                        //if (source.TrimEnd().EndsWith(".mtb", StringComparison.CurrentCultureIgnoreCase))
+                        else if (fi.Extension == Helper.MtbFileExtension)
                         {
                             MtbManager manager = new MtbManager();
                             var paths = manager.Parse(source.Tests);
                             foreach (var p in paths)
                             {
-                                testGroup.Add(new TestInfo(p, p, source.Tests,source.Id));
+                                testGroup.Add(new TestInfo(p, p, source.Tests, source.Id));
                             }
                         }
-                        else if (fi.Extension == ".mtbx")
+                        else if (fi.Extension == Helper.MtbxFileExtension)
                         {
                             testGroup = MtbxManager.Parse(source.Tests, _jenkinsEnvVariables, source.Tests);
 
-                            // set the test Id for each test from the group
-                            // this is important for parallel runner
-                            foreach(var testInfo in testGroup)
-                            {
-                                testInfo.TestId = source.Id;
-                            }
+                            // set the test Id for each test from the group, this is important for parallel runner
+                            testGroup?.ForEach(testInfo => testInfo.TestId = source.Id);
                         }
                     }
                 }
-                catch (Exception)
+                catch
                 {
-                    testGroup = new List<TestInfo>();
+                    testGroup = null;

```

Comment:
here you set the `testGroup` to `null` and later the variable is used without testing if it is a `null` which may cause crash